cop4610/main.c

```diff
@@ -92,39 +93,46 @@ int main(int argc, char* argv[]) {
   int producers = atoi(argv[1]);
   int consumers = atoi(argv[2]);
   int numItems = atoi(argv[3]);
+  int minPerConsumer = (producers*numItems)/consumers;
+  int numExtraItems = (producers*numItems)%consumers;
 
   for(i = 0; i < producers; i++) {
     if((producerChildren[i] = fork()) == 0) { //child producers
 
       int j;
-      for(j = 0; j < atoi(argv[3]); j++) {
+      for(j = 0; j < numItems; j++) {
         synchronizedAccess(&produce, true, true);
       }
       exit(0);
     }
   }
 
-  for(i = 0; i < atoi(argv[2]); i++) {
+  for(i = 0; i < consumers; i++) {
     if((consumerChildren[i] = fork()) == 0) {
       // child
-      int j ;
-      for(j = 0; j < producers*numItems; j++) {
+      for(int j = 0; j < minPerConsumer; j++) {
```

Comment:
the osprey c compiler auto uses c11 to compile but i should have included that in the makefile